src/system-fuel-flow-workspace/system-fuel-flow.service.ts

```diff
@@ -136,53 +136,42 @@ export class SystemFuelFlowWorkspaceService {
     userId: string,
     trx?: EntityManager,
   ) {
-    return new Promise(resolve => {
-      (async () => {
-        const promises = [];
-        for (const fuelFlow of systemFuelFlows) {
-          promises.push(
-            new Promise(innerResolve => {
-              (async () => {
-                const innerPromises = [];
-                const fuelFlowRecord = await withTransaction(
-                  this.repository,
-                  trx,
-                ).getFuelFlowByBeginOrEndDate(sysId, fuelFlow);
-
-                if (fuelFlowRecord) {
-                  innerPromises.push(
-                    await this.updateFuelFlow({
-                      fuelFlowId: fuelFlowRecord.id,
-                      payload: fuelFlow,
-                      locationId,
-                      userId,
-                      isImport: true,
-                      trx,
-                    }),
-                  );
-                } else {
-                  innerPromises.push(
-                    await this.createFuelFlow({
-                      monitoringSystemRecordId: sysId,
-                      payload: fuelFlow,
-                      locationId,
-                      userId,
-                      isImport: true,
-                      trx,
-                    }),
-                  );
-                }
-
-                await Promise.all(innerPromises);
-                innerResolve(true);
-              })();
+    return Promise.all(
+      systemFuelFlows.map(async fuelFlow => {
+        const innerPromises = [];
+        const fuelFlowRecord = await withTransaction(
+          this.repository,
+          trx,
+        ).getFuelFlowByBeginOrEndDate(sysId, fuelFlow);
+
+        if (fuelFlowRecord) {
+          innerPromises.push(
+            await this.updateFuelFlow({
+              fuelFlowId: fuelFlowRecord.id,
+              payload: fuelFlow,
+              locationId,
+              userId,
+              isImport: true,
+              trx,
+            }),
+          );
+        } else {
+          innerPromises.push(
+            await this.createFuelFlow({
+              monitoringSystemRecordId: sysId,
+              payload: fuelFlow,
+              locationId,
+              userId,
+              isImport: true,
+              trx,
```

Comment:
The `innerPromises` could be removed here. It's harmless, though, so not really a necessary change:
```suggestion
    return Promise.all(
      systemFuelFlows.map(async fuelFlow => {
        const fuelFlowRecord = await withTransaction(
          this.repository,
          trx,
        ).getFuelFlowByBeginOrEndDate(sysId, fuelFlow);

        if (fuelFlowRecord) {
          await this.updateFuelFlow({
            fuelFlowId: fuelFlowRecord.id,
            payload: fuelFlow,
            locationId,
            userId,
            isImport: true,
            trx,
          });
        } else {
          await this.createFuelFlow({
            monitoringSystemRecordId: sysId,
            payload: fuelFlow,
            locationId,
            userId,
            isImport: true,
            trx,

```